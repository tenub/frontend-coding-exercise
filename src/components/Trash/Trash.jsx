import React from 'react';
import { DropTarget } from 'react-dnd';
import { TrashAlt } from '@styled-icons/boxicons-solid';

import { ItemTypes } from '../../ItemTypes';

class Trash extends React.PureComponent {
  static types = [
    ItemTypes.BUTTON,
    ItemTypes.INPUT,
    ItemTypes.TEXT,
  ]

  static spec = {
    drop: (props, monitor) => {
      const drop = monitor.getDropResult();
      if (!drop) {
        return;
      }

      const item = monitor.getItem();
      if (typeof item.index === 'undefined') {
        return;
      }

      props.deleteItem(item.index);

      return { name: 'Trash' };
    },
  }

  static collect = (connect, monitor) => ({
    canDrop: monitor.canDrop(),
    connectDropTarget: connect.dropTarget(),
    isOver: monitor.isOver(),
    isOverCurrent: monitor.isOver({ shallow: true }),
    itemType: monitor.getItemType(),
  })

  render() {
    const { isOver, isVisible, connectDropTarget } = this.props;
    const backgroundColor = isOver ? 'lightgrey' : 'white';
    const opacity = isVisible ? 1.0 : 0.0;

    return connectDropTarget(
      <div
        className='element-trash'
        style={{ backgroundColor, opacity }}
      >
        <TrashAlt className='element-trash__icon' />
      </div>
    );
  }
}

export default DropTarget(
  Trash.types,
  Trash.spec,
  Trash.collect,
)(Trash);
