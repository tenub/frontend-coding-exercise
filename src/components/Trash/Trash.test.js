import React from 'react';
import { HTML5Backend } from 'react-dnd-html5-backend';
import { DndProvider } from 'react-dnd';
import TestRenderer from 'react-test-renderer';

import Trash from './';

test('Trash renders successfully', () => {
  const deleteItem = () => {}
  const trash = (
    <DndProvider backend={HTML5Backend}>
      <Trash
        deleteItem={deleteItem}
        isVisible
        items={[]}
      />
    </DndProvider>
  );
  const testRenderer = TestRenderer.create(trash);
  const testTree = testRenderer.toJSON();
  expect(testTree).toMatchSnapshot();
});
