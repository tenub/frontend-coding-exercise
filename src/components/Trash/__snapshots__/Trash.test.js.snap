// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trash renders successfully 1`] = `
<div
  className="element-trash"
  style={
    Object {
      "backgroundColor": "white",
      "opacity": 1,
    }
  }
>
  <svg
    aria-hidden="true"
    className="StyledIconBase-ea9ulj-0 bsimgx element-trash__icon"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 7H5v13a2 2 0 0 0 2 2h10a2 2 0 0 0 2-2V7H6zm10.618-3L15 2H9L7.382 4H3v2h18V4z"
    />
  </svg>
</div>
`;
